smart-contracts/contracts/BridgeBank/BridgeBank.sol

```diff
@@ -548,18 +548,24 @@ contract BridgeBank is BankStorage, CosmosBank, EthereumWhiteList, CosmosWhiteLi
     require(token.length == amount.length, "M_P");
     require(token.length == isBurn.length, "M_P");
 
-    uint256 intermediateLockBurnNonce = lockBurnNonce;
+    uint256 recipientLength = recipient.length;
+
+    // lockBurnNonce contains the previous nonce that was
+    // sent in the LogLock/LogBurn, so the first one we send
+    // should be lockBurnNonce + 1
+    uint256 startingLockBurnNonce = lockBurnNonce + 1;
```

Comment:
If i execute a reentrancy at _burn/_lockTokens, would I be able to emit an event with nonce that is ahead, such that nonce is no longer strictly increasing? If so it would halt the bridge I believe